src/ast/ast_validations.c

```diff
@@ -54,7 +54,22 @@ static void _AST_GetIdentifiers(const cypher_astnode_t *node, rax *identifiers)
 	 * @12  20..23  > > > > > function name    `max`
 	 * @13  24..25  > > > > > identifier       `z`
 	 * @14  20..26  > > > > identifier         `max(z)` */
-	if(cypher_astnode_type(node) == CYPHER_AST_PROJECTION) child_count = 1;
+	cypher_astnode_type_t type = cypher_astnode_type(node);
+	if(type == CYPHER_AST_PROJECTION) child_count = 1;
+
+	if(type == CYPHER_AST_LIST_COMPREHENSION) {
+		// A list comprehension has a local variable that should only be accessed within its scope.
+		// Temporarily add it to the rax, then remove it after validating the other elements.
+		const cypher_astnode_t *variable_node = cypher_ast_list_comprehension_get_identifier(node);
+		const char *variable = cypher_ast_identifier_get_name(variable_node);
+		raxInsert(identifiers, (unsigned char *)variable, strlen(variable), NULL, NULL);
```

Comment:
I don't see the purpose in adding `variable` to `identifiers` calling `_AST_GetIdentifiers` recessively and then removing it. why is that done?

Reply:
I want to perform validations on the internals of the list comprehension, where the `variable` alias can validly be referenced, but ensure that outside of the list comprehension's scope `variable` cannot be referenced.

Reply:
Oh, you're correct - my bad! Will remove the insertion.